bokeh/sphinxext/bokeh_plot.py

```diff
@@ -235,13 +237,17 @@ def _show(obj=None):
 io.show = _show
 
 
-def _render_plot(source, symbol):
+def _render_plot(source, path=None, symbol=None):
     io._state._document = Document()
     namespace = {}
+    if path is not None:
+        sys.modules["fake"] = types.ModuleType("fake")
+        sys.modules["fake"].__file__ = path
+        namespace["__name__"] = "fake"
```

Comment:
Roughly seems OK, I'd suggest building the docs and spot checking plots in various places (and all the extension ones). 

As an aside I think almost all of this complicated code can be ripped out and replaced with a subprocess call to `bokeh json` and some trivial templating.